lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -0,0 +1,21 @@
+"""
+This file contains implementation override of SearchFilterGenerator which will allow
+    * Filter by all courses in which the user is enrolled in
+"""
+
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from search.filter_generator import SearchFilterGenerator
+
+from lms.djangoapps.courseware.courses import get_courses
+
+
+class LmsSearchFilterGenerator(SearchFilterGenerator):
+    """ SearchFilterGenerator for LMS Search """
+
+    def field_dictionary(self, **kwargs):
+        """ add course if provided otherwise add courses in which the user is enrolled in """
+        field_dictionary = super(LmsSearchFilterGenerator, self).field_dictionary(**kwargs)
+        if "course_id" not in kwargs:
+            field_dictionary['courses'] = [str(course.id) for course in get_courses(kwargs['user'])]
```

Comment:
You may want to use the name `field_dictionary['course']` - were you able to confirm if the array in the filter has the desired effect?